flink-runtime/src/test/java/org/apache/flink/runtime/taskexecutor/TaskExecutorPartitionLifecycleTest.java

```diff
@@ -206,8 +196,7 @@ private void testJobMasterConnectionTerminationAfterExternalReleaseOrPromotion(T
 			// once this returns we know that the TE has exited releasePartitions and associated connection checks
 			runInTaskExecutorThreadAndWait(taskExecutor, () -> assertTrue(jobManagerTable.contains(jobId)));
 
-			final CompletableFuture<Collection<ResultPartitionID>> secondReleasePartitionsCallFuture = new CompletableFuture<>();
-			runInTaskExecutorThreadAndWait(taskExecutor, () -> shuffleEnvironment.releasePartitionsLocallyFuture = secondReleasePartitionsCallFuture);
+			runInTaskExecutorThreadAndWait(taskExecutor, () -> partitionTracker.setIsTrackingPartitionsForFunction(jobId -> false));
```

Comment:
Instead of updating the `isTrackingPartitionsForFunction` one could also define/set a function in the beginning which returns an `AtomicBoolean` which one could use to trigger the behaviour change.